src/ddscxx/src/org/eclipse/cyclonedds/domain/DomainParticipantDelegate.cpp

```diff
@@ -117,6 +117,66 @@ org::eclipse::cyclonedds::domain::DomainParticipantDelegate::DomainParticipantDe
     }
 }
 
+org::eclipse::cyclonedds::domain::DomainParticipantDelegate::DomainParticipantDelegate(
+  uint32_t id,
+  const dds::domain::qos::DomainParticipantQos & qos,
+  dds::domain::DomainParticipantListener * listener,
+  const dds::core::status::StatusMask & event_mask,
+  const ddsi_config & config
+)
+  : domain_id_(id), qos_(qos)
+{
+    org::eclipse::cyclonedds::domain::DomainWrap::map_ref_iter it;
+    dds_qos_t * ddsc_qos;
+    dds_domainid_t did;
+    dds_return_t ret;
+
+    if ((id != org::eclipse::cyclonedds::domain::default_id()) && (id > 230)) {
+      ISOCPP_THROW_EXCEPTION(ISOCPP_INVALID_ARGUMENT_ERROR, "Invalid domain_id: %d", (int)id);
```

Comment:
Old (C-style) cast used here and at some following lines.
It would be better to use static_cast<int> to be more in line with the rest of the code.